src/node_url.cc

```diff
@@ -705,10 +710,11 @@ namespace url {
             p++;
             continue;
           } else if (ch == ':' || (has_state_override && ch == kEOL)) {
```

Comment:
As long as we're still API compliant and passing all the tests, such deviation is fine. It would be helpful, however, to document the deviations in code comments.